include/deal.II/fe/fe_values_coupling.h

```diff
@@ -0,0 +1,557 @@
+     */
+    using value_type = typename ViewType::value_type;
+
+    /**
+     * An alias for the data type of gradients of the underlying view.
+     */
+    using gradient_type = typename ViewType::gradient_type;
+
+    /**
+     * An alias for the data type of the product of a @p Number and the values
+     * of the underlying view type.
+     */
+    template <typename Number>
+    using solution_value_type = typename ProductType<Number, value_type>::type;
+
+    /**
+     * An alias for the data type of the product of a @p Number and the gradients
+     * of the underlying view type.
+     */
+    template <typename Number>
+    using solution_gradient_type =
+      typename ProductType<Number, gradient_type>::type;
+
+    /**
+     * Construct a new ReorderedView object.
+     *
+     * The @p dof_renumbering vector is used to renumber the degrees of freedom,
+     * while the @p quadrature_renumbering vector is used to renumber the
+     * quadrature points. An empty renumbering vector simply means that no
+     * renumbering is performed.
+     *
+     * @note The renumbering vectors may *not* match the dimensions of the
+     * underlying view, i.e., you could use this class to only run over half of
+     * the degrees of freedom, and integrating only over half of the current
+     * cell, if you wish to do so. The important part is that every entry of
+     * each renumbering vector is a legal value for the underlying view object.
+     * We allow the dof renumbering vector to contain
+     * `numbers::invalid_unsigned_int` values, which are ignored, and produce
+     * zero values, gradients, hessians, etc. for the corresponding shape
+     * function.
+     *
+     * @param view The underlying FEValuesViews object.
+     * @param dof_renumbering A renumbering vector for the degrees of freedom.
+     * @param quadrature_renumbering A renumbering vector for the quadrature
+     * points.
+     */
+    ReorderedView(const ViewType &                 view,
+                  const std::vector<unsigned int> &dof_renumbering        = {},
+                  const std::vector<unsigned int> &quadrature_renumbering = {});
+
+    /**
+     * Return the value of the underlying view, for the shape function and
+     * quadrature point selected by the arguments.
+     *
+     * @param shape_function Number of the shape function to be evaluated. Note
+     * that this number runs from zero to size of the renumbering vector
+     * provided at construction time, or to `dofs_per_cell`, if the renumbering
+     * vector is empty.
+     *
+     * @param q_point Number of the quadrature point at which the function is to
+     * be evaluated. Note that this number runs from zero to the size of the
+     * renumbering vector provided at construction time, or to
+     * `n_quadrature_points`, if the renumbering vector is empty.
+     *
+     * @dealiiRequiresUpdateFlags{update_values}
+     */
+    typename ViewType::value_type
```

Comment:
```suggestion
    value_type
```
Same below.

Reply:
Or if you haven't already defined a `value_type`, then add the `using` declaration to introduce it.